tests/unit/moduleapi/timer.tcl

```diff
@@ -54,7 +56,23 @@ start_server {tags {"modules"}} {
         assert_equal {} [r test.gettimer $id]
     }
 
-    test "Unload the module - timer" {
+    test "Module can be unloaded only when timer was finished" {
+        r set "timer-incr-key" 0
+        set id [r test.createtimer 2000 timer-incr-key]
+
+        # Make sure the Timer has not been fired
+        assert_equal 0 [r get timer-incr-key]
+        # Module can not be unload since the timer was ongoing 
+        catch [r module unload timer] err
+        assert_match {ERR*} $err
+
+        # Wait to be sure timer has been finished
+        wait_for_condition 10 1000 {
+            [r get timer-incr-key] == 1
+        } else {
+            fail "Timer not fired"
+        }
```

Comment:
Maybe we don't need to do timer-incr-key verification in this test, because it already does the same thing in other tests.
This test is supposed to verify when the timer has not finished and then do a module unload.
I think the steps should be:
1. start a 5 sec timer
2. check module unload fail
3. stop the timer.
4. check module unload success.

Reply:
yes, the verification is to make sure the timer has been fired as expected, so it can do the next step. And you have mentioned that to stop the timer as the precondition, I will add in the later commit.